### PR TITLE
feat(form/builder): add hidden operator and fix default case typo

### PR DESCRIPTION
**Pull request by jmanrumartinez**
Adds **HIDDEN** operator and fixes a typo

**How to use**
As a normal conditional, you should declare which field is checked.

```
{
      when: [
        {
          operator: 'EQUALS',
          id: FIELD_KEYS.WANT_TO_ADD_SECONDARY_PHONE,
          value: ['false']
        },
        {
          operator: 'NIN',
          id: FIELD_KEYS.MAIN_PHONE,
          value: ['']
        },
        {
          operator: 'HIDDEN',
          id: FIELD_KEYS.SECONDARY_PHONE
        }
      ],
      then: {
        data: {
          hidden: false
        }
      }
}
```

The rule will trigger if the `WANT_TO_ADD_SECONDARY_PHONE` value is `false`, there're value in `MAIN_PHONE` and the `SECONDARY_PHONE` is hidden. 

**Why it's needed and when to use**
In our case, we only want to show the wantToAddSecondaryPhone field if the `secondaryPhone` field is hidden, because there're cases like the `secondaryPhone` field is saved in localStorage and initially appears the two fields (main phone and secondary phone)

Captura de pantalla 2021-09-02 a las 11 25 22

So without the hidden condition will trigger because `WANT_TO_ADD_SECONDARY_PHONE` is initialized with false value, and there's value in the `MAIN_PHONE` (it will appear the `WANT_TO_ADD_SECONDARY_PHONE` field if the user rewrites the phone)

### CODE REVIEW

File: components/form/builder/src/reducer/operators.js [hunk 1]
```diff
@@ -104,5 +105,9 @@ export const operators = {
     const isEmpty = !fieldValue.length
     const hasNotExpectedValues = values.some(i => !fieldValue.includes(i))
     return isEmpty || hasNotExpectedValues
+  },
+  [HIDDEN]: (id, fields) => {
+    const {hidden} = pickFieldById(fields, id)
+    return hidden === true
```
Comment (jmanrumartinez) [inline components/form/builder/src/reducer/operators.js:111]: Do you prefer to return hidden directly?

Comment (arnau-rius) [reply components/form/builder/src/reducer/operators.js:111]: yes

Comment (arnau-rius) [reply components/form/builder/src/reducer/operators.js:111]: If you want to make sure it is a boolean you can cast it using `Boolean(hidden)` or `!!hidden`

File: components/form/builder/src/reducer/rules.js [hunk 2]
```diff
@@ -105,8 +107,11 @@ export const shouldApplyRule = (fields, changeField, locale) => when => {
       case NSUPERSET:
         isValid = operators.NSUPERSET(rule.id, rule.value, fields, locale)
         break
+      case HIDDEN:
+        isValid = operators.HIDDEN(rule.id, fields)
+        break
       default:
-        console.warn(`Unkown operator ${rule.operator}`)
+        console.warn(`Unknown operator ${rule.operator}`)
```
Comment (arnau-rius) [inline components/form/builder/src/reducer/rules.js:114]: ✅ speller

**Review by arnau-rius — APPROVED** [review]

File: components/form/builder/src/reducer/operators.js (hunk 1 above)
Comment (andresz1) [reply components/form/builder/src/reducer/operators.js:111]: ⬆️ agree

**Comment by andresz1** [comment]
IMO documentation + example would be awesome and why is the operator needed!

**Review by tomasmax — APPROVED** [review]

**Comment by jmanrumartinez** [comment]
> IMO documentation + example would be awesome and why is the operator needed!

Updated PR with some explanation and updated the docs [here](https://github.mpi-internal.com/scmspain/all--lib-form-builder-docs/pull/13)

**Review by andresz1 — APPROVED** [review]

**Pull request merged**